docs/content/develop/resource.md

```diff
@@ -0,0 +1,455 @@
+   # immutable: true
+
+   # Overrides one or more timeouts, in minutes. All timeouts default to 20.
+   # timeouts: !ruby/object:Api::Timeouts
+   #   insert_minutes: 20 
+   #   update_minutes: 20 
+   #   delete_minutes: 20 
+
+   # URL for creating a new resource, including query parameters.
+   # Terraform field names enclosed in double curly braces will be replaced
+   # with the field values from the resource.
+   create_url: 'projects/{{project}}/locations/{{location}}/resourcenames?resourceId={{resource_id}}'
+   # Overrides the HTTP verb used to create a new resource.
+   # Allowed values: :POST, :PUT, :PATCH. Default: :POST
+   # create_verb: :POST
+
+   # Overrides the update URL for the resource. (Otherwise, the self_link URL
+   # will be used.)
+   # Terraform field names enclosed in double curly braces will be replaced
+   # with the field values from the resource.
+   # update_url: 'projects/{{project}}/locations/{{location}}/resourcenames/{{resource_id}}'
+   # The HTTP verb used to update a resource. Allowed values: :POST, :PUT, :PATCH. Default: :PUT.
+   update_verb: :PATCH
+   # True if the resource should use an update mask for updates.
+   update_mask: true
+
+   # Overrides the delete URL for the resource. (Otherwise, the self_link URL
+   # will be used.)
+   # Terraform field names enclosed in double curly braces will be replaced
+   # with the field values from the resource.
+   # delete_url: 'projects/{{project}}/locations/{{location}}/resourcenames/{{resource_id}}'
+   # Overrides the HTTP verb used to delete a resource.
+   # Allowed values: :POST, :PUT, :PATCH, :DELETE. Default: :DELETE
+   # delete_verb: :DELETE
+
+   # Enables generation of code to handle API calls that return operations.
+   autogen_async: true
+   # Sets parameters for handling operations returned by the API.
+   async: !ruby/object:Api::OpAsync
+     # Overrides which API calls return operations. Default: ['create',
+     # 'update', 'delete']
+     # actions: ['create', 'update', 'delete']
+     operation: !ruby/object:Api::OpAsync::Operation
+       base_url: '{{op_id}}'
+     # If true, the completed operation's returned JSON will be expected to
+     # contain a full resource in the "response" field
+     # result: !ruby/object:Api::OpAsync::Result
+     #   resource_inside_response: true
+
+   # All resources (of all kinds) that share a mutex value will block rather
+   # than executing concurrent API requests.
+   # Terraform field names enclosed in double curly braces will be replaced
+   # with the field values from the resource.
+   # mutex: RESOURCE_NAME/{{resource_id}}
+
+   # IAM_GOES_HERE
+
+   # EXAMPLES_GO_HERE
+
+   parameters:
+     - !ruby/object:Api::Type::String
+       name: 'location'
+       required: true
+       immutable: true
+       url_param_only: true
+       description: |
+         LOCATION_DESCRIPTION
+     - !ruby/object:Api::Type::String
+       name: 'resource_id'
```

Comment:
`resource_id` is used zero times in our codebase, let's avoid introducing new practices that deviate from everything else in a documentation update. The most common pattern is a `name` field that's a `url_param_only`.

Reply:
`resource_id` is a common pattern in our codebase - for example, AlloyDB cluster uses `cluster_id` rather than `name`. https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/alloydb/Cluster.yaml `name` introduces potential for confusion because it's short on the Terraform side and long on the API side. That said, it is an older pattern and likely more common so I'd be okay with sticking with it for now.

Reply:
Switched to name in fbefc0fad.

Reply:
Ah- you meant `{{resource}}_id` and not the literal `resource_id`. Yeah, that's been used a few times. IMO we should prefer `name` as `url_param_only`, as that's consistent with GCE and the most other resources. It's unfortunate that aip.dev/122 chose `name` as the name for what's conventionally called an id (such as our id field) and `id` to describe what's conventionally a name.